### PR TITLE
US1 – List of Outivities

### PR DESCRIPTION
**Pull request by Julia-Pickel**
**In the 1st User Story we did following tasks:**

- Create feature-branch
- Create example data with 6 Outivities
- Create and display headline with the title *"All Outivities"*
- Create Card Component
- Create List Component
- Render Cards in List Component
- Render List in Homepage
- Add error handling
- and we added some styling

We are looking forward to getting some feedback on that. – Thanks in advance ! 🙌

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -1,7 +1,80 @@
...
+  background-color: var(--primary-color);
+  color: var(--neutral-color);
+`;
+
+const StyledOutivitieCardContainer = styled.ul`
+  display: grid;
+  gap: 1rem;
+  padding: 20px;
+  place-content: center;
+  gap: 50px;
```
Comment (JessicaLoers) [inline pages/index.js:51]: Redundant 'gap' property declaration. Consider removing one. 🧹

File: pages/index.js [hunk 2]
```diff
@@ -1,7 +1,80 @@
...
+import Image from "next/image";
+import styled from "styled-components";
+
 export default function HomePage() {
-  return (
-    <div>
-      <h1>Hello from Next.js</h1>
-    </div>
-  );
+  try {
```
Comment (JessicaLoers) [inline pages/index.js]: A `try...catch` block in the render function is not necessary in this context.

**Review by JessicaLoers — CHANGES_REQUESTED** [review]
**Hello Team Outivities** -  congrats to your first PR. ✨

I have a few suggestions  for improvement:

- Since the data `outivities` is statically imported, runtime errors related to data fetching are unlikely, beacuse React rendering logic usually doesn't throw synchronous `errors `that `try...catch `would catch.

➡️  _Sidenote_ These `errors` are more likely to come from asynchronous operations like API calls.

- Consider adding more descriptive semantic elements, for example a [`<article>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/article?retiredLocale=de) for the outivity card and a `<main>`  which encapsulate the primary content that is unique to that page.

**Otherwise well done.**  🙂

**Comment by Julia-Pickel** [comment]
Hey @JessicaLoers, thank you for your recommendations !!

We updated the User Story 1 as follows:
- removed `try... catch` 
- removed double gap of `ul` named *StyledOutivityCardContainer*
- html semantic: 
    - added an `<article>` framing our `ul`
    - added a `<main>` framing our article
    - added a `<Head>` for the HomePage

File: pages/index.js [hunk 3]
```diff
@@ -1,7 +1,90 @@
...
+        <main>
+          <h1>Ooops... </h1>
+          <p>
+            Error Loading Outivities. We were unable to load the Outivities
+            list. Please try again later.
+          </p>
+        </main>
+      </>
+    );
+  }
```
Comment (Julia-Pickel) [inline pages/index.js]: We still have kind of an error handling as it was part of our tasks of the User Story.
As we are directly importing it from a file we are quite safe here but we are not sure if we can fully drop this for now?

Comment (JessicaLoers) [reply pages/index.js]: Yes you can fully drop this 🚀

**Comment by lennart-kaminsky** [comment]
Hej @Julia-Pickel, @BirkBee , @ykolomytseva and @Philipp-Hess!

It's really cool and interesting to see your start into this project. Really nice way to design, we also learned a lot from reading your code.

We don't really have anything to add to @JessicaLoers review - it seems that you just have to drop the `return` with the error handling and you're good to go!

**Comment by Julia-Pickel** [comment]
Hi @lennart-kaminsky !

Thank you for your feedback and kind words 🙌
We removed the error handling :)

**Comment by lennart-kaminsky** [comment]
LGTM

**Review by JessicaLoers — APPROVED** [review]
Yeahhhhhh. 🚀

**Pull request merged**